cmd/containerboot/main.go

```diff
@@ -195,18 +195,21 @@ func run() error {
 		return fmt.Errorf("failed to bring up tailscale: %w", err)
 	}
 	killTailscaled := func() {
+		// The default termination grace period for a Pod is 30s. We wait 25s at
+		// most so that we still reserve some of that budget for tailscaled
+		// to receive and react to a SIGTERM before the SIGKILL that k8s
+		// will send at the end of the grace period.
+		ctx, cancel := context.WithTimeout(context.Background(), 25*time.Second)
+		defer cancel()
+
+		if err := ensureServicesNotAdvertised(ctx, client); err != nil {
```

Comment:
This might be a little difficult to find for someone looking for ingress ProxyGroup lifecycle flow. Maybe it's worth to add a note about this logic near where we are setting egress lifecycle hook https://github.com/tailscale/tailscale/blob/6d117d64a256234372f2bb177392b987aa1758af/cmd/k8s-operator/proxygroup_specs.go#L202

I think it would also make sense to add a note as to why we chose to not use prestop hook for this in the same place as it might be non-obvious

Reply:
Might also make sense to add a note about possible readiness gate in future if needed

Reply:
Good shout 👍 done